### PR TITLE
Implemented workaround for Bug in ClassificationStore booleanSelect

### PR DESCRIPTION
**Pull request by woellundwoell**
## General preparations

- Create a key definition of type "booleanSelect" in the classification store
- Assign the keyDefintion to a group
- Assign the classification store to an class attribute
- Create an object of this class
- Assign the above created group in the classification store attribute
- Assign all possible values ​​(No / Yes / Empty) to the booleanSelect attribute

## Expected behavior

- Correct values are saved
- Display the correct values ​​after reload

## Actual Behavior

- Only saves the "Yes value" correctly
- The values ​​"No" and "Empty" are not stored in the database
- Even after manually setting the correct values ​​in the database (-1 for "No", and NULL for "Empty"), the values ​​are not displayed correctly

## Workaround

As discussed we implemented a workaround for this bug and inserted a TODO for a final fix

### CODE REVIEW

File: pimcore/models/DataObject/ClassDefinition/Data/BooleanSelect.php [hunk 1]
```diff
@@ -270,7 +270,8 @@ public function checkValidity($data, $omitMandatoryCheck = false)
      */
     public function isEmpty($data)
     {
-        return $data !== true && $data !== false;
+    	// For all 3 states never empty
+        return false;
```
Comment (weisswurstkanone) [inline pimcore/models/DataObject/ClassDefinition/Data/BooleanSelect.php]: What is the reason for this change? Will this break the inheritance?

**Comment by woellundwoell** [comment]
The reason is that the booleanSelect needs a data set (in DB) for each state - also for NULL / EMPTY.
If the isEmpty function returns true for the NULL value, the dataset will be deleted. This has the effect that the inheritance does not work and in the GUI  an empty Select is displayed instead of the "empty value".

**Comment by weisswurstkanone** [comment]
thx!
